mesonbuild/backend/ninjabackend.py

```diff
@@ -3586,8 +3586,9 @@ def generate_scanbuild(self) -> None:
             return
         if 'scan-build' in self.all_outputs:
             return
+        subprojdir = os.path.join(self.environment.get_source_dir(), self.build.get_subproject_dir())
```

Comment:
I think we can simplify this by only writing the source directory once into build.ninja, and then joining it inside scanbuild.py

Reply:
Good point! This has been implemented.